[2024-11-27 23:22:40,700][root][INFO] - Initializing Model
[2024-11-27 23:23:21,508][root][INFO] - Model initialized in 40.81 seconds
[2024-11-27 23:23:21,509][root][INFO] - Loading Datasets
[2024-11-27 23:23:31,820][root][INFO] - Datasets loaded in 10.31 seconds
[2024-11-27 23:23:31,987][root][INFO] - Starting Training
[2024-11-28 01:01:11,593][root][INFO] - Evaluating at the end of epoch
[2024-11-28 01:01:11,957][root][ERROR] - An error occurred: NRMS.forward() got an unexpected keyword argument 'candidate_news_id'
Traceback (most recent call last):
  File "/home/shouju/rsch/news-recommendation-llm/src/experiment/train.py", line 183, in train
    trainer.train()
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/transformers/trainer.py", line 2123, in train
    return inner_training_loop(
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/transformers/trainer.py", line 2572, in _inner_training_loop
    self.control = self.callback_handler.on_epoch_end(args, self.state, self.control)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/transformers/trainer_callback.py", line 478, in on_epoch_end
    return self.call_event("on_epoch_end", args, state, control)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/transformers/trainer_callback.py", line 518, in call_event
    result = getattr(callback, event)(
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/rsch/news-recommendation-llm/src/experiment/train.py", line 30, in on_epoch_end
    metrics = evaluate(kwargs['model'], self.eval_dataset, self.device)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/rsch/news-recommendation-llm/src/experiment/train.py", line 56, in evaluate
    model_output: ModelOutput = net(**batch)
                                ^^^^^^^^^^^^
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/accelerate/utils/operations.py", line 823, in forward
    return model_forward(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/accelerate/utils/operations.py", line 811, in __call__
    return convert_to_fp32(self.model_forward(*args, **kwargs))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/shouju/anaconda3/envs/rec-llm/lib/python3.11/site-packages/torch/amp/autocast_mode.py", line 44, in decorate_autocast
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
TypeError: NRMS.forward() got an unexpected keyword argument 'candidate_news_id'
